toolchain/check/merge.cpp

```diff
@@ -268,22 +268,99 @@ static auto CheckRedeclParams(Context& context, SemIRLoc new_decl_loc,
   return true;
 }
 
+// Returns true if the two nodes represent the same syntax.
+static auto IsNodeSyntaxEqual(Context& context, Parse::NodeId new_node_id,
+                              Parse::NodeId prev_node_id) -> bool {
+  if (context.parse_tree().node_kind(new_node_id) !=
+      context.parse_tree().node_kind(prev_node_id)) {
+    return false;
+  }
+
+  // TODO: Should there be a trivial way to check if we need to check spellings?
+  // Identifiers and literals need their text checked, but keywords and
```

Comment:
(It'd be nice to just compare the `IdentifierId`s for identifiers where possible, though we'd need to check for `r#` somehow and still presumably need string comparisons for cross-file cases. But not in this PR!)

Reply:
Adjusted